### PR TITLE
Remove 4 echo instances from security tests

### PR DESCRIPTION
**Pull request by howardjohn**
For https://github.com/istio/istio/issues/29475

### CODE REVIEW

File: tests/integration/security/pass_through_filter_chain_test.go [hunk 1]
```diff
@@ -39,240 +37,308 @@ func TestPassThroughFilterChain(t *testing.T) {
...
+			}
+
+			for _, cluster := range ctx.Clusters() {
+				client := apps.Naked.Match(echo.InCluster(cluster)).GetOrFail(ctx, echo.Namespace(ns.Name()))
+				destination := apps.A.Match(echo.Namespace(ns.Name())).GetOrFail(ctx, echo.InCluster(cluster))
+				from := getWorkload(client, ctx)
+				destWorkload := getWorkload(destination, ctx).Address()
+				for _, tc := range cases {
+					ctx.NewSubTest(fmt.Sprintf("In %s/%v", cluster.StableName(), tc.name)).Run(func(ctx framework.TestContext) {
+						ctx.Config().ApplyYAMLOrFail(ctx, ns.Name(), tc.config)
```
Comment (yangminzhu) [inline tests/integration/security/pass_through_filter_chain_test.go:291]: How do you make sure config applied in the last test does not affect the current test case? Same question for the changes to the JWT tests, one benefit of applying configs to multiple instances is that it makes sure any past configs will not affect next tests.

Sometimes the test might be passing due to left-over configs  from last test and thus causing the current test not reflecting the real situation.

Comment (howardjohn) [reply tests/integration/security/pass_through_filter_chain_test.go:291]: It is possible. However, we have two mitigations. First, we have `retry.Converge(3)` which means we send multiple requests and ensure they all response the same. The other is we run these tests 100s of times per day, so if we do have a regression and it isn't detected once, it is extremely likely to pop up as a flaky test at least.

We do this for other tests and haven't had issues here, but open to suggestions - @stevenctl ? We could add an `istioctl wait` but its not very reliable

One benefit of this is we are gaining coverage of the envoy config changing, which has had a lot of bugs historically.

Comment (stevenctl) [reply tests/integration/security/pass_through_filter_chain_test.go:291]: Other security tests do exactly that:

https://github.com/istio/istio/blob/6d5406628e2fa9bd2fa31e9b4bc5d05186e01bd6/tests/integration/security/util/reachability/context.go#L103

https://github.com/istio/istio/blob/6d5406628e2fa9bd2fa31e9b4bc5d05186e01bd6/pkg/test/framework/components/istioctl/kube.go#L61

**Review by stevenctl — APPROVED** [review]

File: tests/integration/security/pass_through_filter_chain_test.go [hunk 2]
```diff
@@ -39,240 +37,308 @@ func TestPassThroughFilterChain(t *testing.T) {
...
-								} else {
-									// Check HTTP forbidden response
-									if len(responses) >= 1 && response.StatusCodeForbidden == responses[0].Code {
-										return nil
-									}
+							}
+							ctx.NewSubTest(name).Run(func(ctx framework.TestContext) {
+								retry.UntilSuccessOrFail(ctx, func() error {
+									responses, err := from.ForwardEcho(context.TODO(), request)
+									log.Errorf("howardjohn: got %v for %v", err, request.Url)
```
Comment (yangminzhu) [inline tests/integration/security/pass_through_filter_chain_test.go:310]: the log needs update.

File: tests/integration/security/pass_through_filter_chain_test.go [hunk 3]
```diff
@@ -39,240 +37,308 @@ func TestPassThroughFilterChain(t *testing.T) {
...
+			cases := []struct {
+				name     string
+				config   string
+				expected []expect
+			}{
+				// There is no authN/authZ policy.
+				// All requests should success, this is to verify the pass through filter chain and
+				// the workload ports are working correctly.
+				{
+					"DISABLE",
```
Comment (yangminzhu) [inline tests/integration/security/pass_through_filter_chain_test.go:54]: can you add the field name explicitly?

File: tests/integration/security/pass_through_filter_chain_test.go (hunk 1 above)
Comment (yangminzhu) [reply tests/integration/security/pass_through_filter_chain_test.go:291]: Can we also use `WaitForConfigWithSleep` here and jwt_test.go? the retry may not always help when the current test passed for the first time due to effect from last test, in this case the retry won't be triggered.

**Review by yangminzhu — COMMENTED** [review]
looks good, just some small comments.

File: tests/integration/security/pass_through_filter_chain_test.go (hunk 1 above)
Comment (yangminzhu) [reply tests/integration/security/pass_through_filter_chain_test.go:291]: By the way, did we use `retry.Converge()` here? I only find the `DefaultCallRetryOptions()`.

**Comment by howardjohn** [comment]
@yangminzhu I cleaned up the log, added field names, and added the explicit wait

**Review by yangminzhu — APPROVED** [review]
Thanks!

**Pull request merged**